File: tutorial6/app/src/main/java/com/rudeo/tutorial6/MainActivity.java
package com.rudeo.tutorial6;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.EditText;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
    }

    public void moveClick(View v)
    {
        EditText editText = (EditText)findViewById(R.id.editText);

        String name = editText.getText().toString();
        Intent intent = new Intent(getApplicationContext(), SubActivity.class);

        intent.putExtra("nameText", name);
        startActivity(intent);
    }

}

File: BMI_calculator2/app/src/main/java/com/rudeo/bmi_calculator/MainActivity.kt
package com.rudeo.bmi_calculator

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.EditText
import android.widget.Toast
import com.rudeo.bmi_calculator.databinding.ActivityMainBinding

class MainActivity : AppCompatActivity() {


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)


    }



    fun calculator(view: android.view.View) {

        val heightEditText : EditText = findViewById(R.id.heightEditText)
        val weightEditText : EditText = findViewById(R.id.weightEditText)


        val height : Int = heightEditText.text.toString().toInt()
        val weight : Int = weightEditText.text.toString().toInt()

        val intent = Intent(view.context, ResultActivity::class.java)

        intent.putExtra("height",height)
        intent.putExtra("weight", weight)
        startActivity(intent)
    }


}